http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java

```diff
@@ -150,6 +164,13 @@ private void releaseParentChannel(Channel parentChannel) {
 
     @Override
     public void close() {
+        connections.forEach(c -> {
```

Comment:
Should these awaits eventually timeout? Also it may not be a good idea to dispatch the close to the connection pool as it also awaits so that would block a netty thread. I think we can safely move it to the calling thread. What do you think?

Reply:
Not sure about having a timeout, but what about switching to a normal await() so it's interruptible? +1 moving the close on to the calling thread.

Reply:
+1 to the interruptible